### ISSUE TITLE
Where's GlobalDefaults.ini under Ubuntu 12.04 64 bits? 

### ISSUE THREAD

**Issue by rossanag** [occipital/OpenNI2#101]
Hi, I decided to upgrade to OpenNI2 from OpenNI. I installed it from the binary.
I have libfreenect updated, I tested through the example and it works fines. All the examples under OpenNI work fine also.
However I want to take advantage of the simplified version by installing OpenNI2.
According to the Quick start Guide I need to change some line in the GlobalDefaults.ini file.
(change UsbInterface=2 to UsbInterface=0)

I am not able to find this file under etc/openni, actually I have no the openni directory under etc.
Actually I have no this file, I have the PS1080.ini instead under the OpenNI.
/home/rossg/Kinect/OpenNI2/OpenNI2/Redist/OpenNI2/Drivers

Any suggestion is highly appreciated.
Best regards.

**Comment by jselikof** [comment]
@rossanag, for Linux use the following line: 
UsbInterface=2
The following may also be necessary:
https://github.com/occipital/OpenNI2/issues/88#issuecomment-225604223

**Comment by rossanag** [comment]
@jslikof thanks for your response, actually I am aware about the changes,
the real problem is that I can't install it, I have no file with this name

**Comment by jselikof** [comment]
Was OpenNi2 installed from the appropriate Linux binary here: http://structure.io/openni ?

**Comment by rossanag** [comment]
I changed this file, no luck, and it isn't where it is supposed to be,

**Comment by jselikof** [comment]
GlobalDefaults.ini isn't used in OpenNi2.  Please try installing OpenNi2 from the link given above.

**Comment by rossanag** [comment]
Sorry my bad, I am from the cell phone. yes, i tryed both and I changed
this file also. Neither the binary nor the pre build version worked.
By the way the documentation refers to GlobalDefaults, ini for Linux, it
should be changed.

As soon as I could try more things I will report them here.
Thanks a lot


El 05/12/2016 13:49, "Rossana Guerra" <guerra.rossana@gmail.com> escribió:

I changed this file, no luck, and it isn't where it is supposed to be,

**Comment by rossanag** [comment]
@jselikof  I tryed everything you suggested, actually I already had the soft link created also,
it doesn't work, when I try to run NiViewer example, I get the following error:

_> 
> openDevice failed:
> 	DeviceOpen using default: no devices found
> 
> Press any key to continue_ 

I am going to follow some new steps, if I succeed I will let you know.

**Comment by jselikof** [comment]
Try running `strace -y ./NiViewer` for additional debug info.

**Comment by rossanag** [comment]
@jselikof I was trying the following changes according to many post on stackoverflow and github.
In short:
- copied the libs files under OpenNI2  to usr//lib
- copied  the libFreenectDriver.so* to /path/to/OpenNI2/Redist
- changed Open.ini under Redist by adding the driver path
- added some environment variables
export OPENNI2_REDIST=/home/studio/OpenNI2/Redist
export OPENNI2_DRIVERS_PATH=/home/studio/OpenNI2/Redist/OpenNI2/Drivers

Now I get the following error everytime I try to run the examples under the Samples directory:
LibraryHandler: Couldn't load library libFreenectDriver.so
```
	DeviceDriver: library handle is invalid for file libFreenectDriver.so
	Couldn't understand file 'libFreenectDriver.so' as a device driver
	LibraryHandler: Couldn't find function oniDriverCreate in libOpenNI2.jni.so. Stopping
	DeviceDriver: library handle is invalid for file libOpenNI2.jni.so
	Couldn't understand file 'libOpenNI2.jni.so' as a device driver
	LibraryHandler: Couldn't find function oniDriverCreate in libOpenNI2.so. Stopping
	DeviceDriver: library handle is invalid for file libOpenNI2.so
	Couldn't understand file 'libOpenNI2.so' as a device driver
	Found no valid drivers in './../../Tools'

However if I try to run NiViewer, again:
```
openDevice failed:
	DeviceOpen using default: no devices found

Press any key to continue . . .

This is the output when executing strace -v ./NiViewer:

[NiViewer.txt](https://github.com/occipital/OpenNI2/files/633115/NiViewer.txt)

Thanks so much

```
```

- set environment variables

**Comment by jselikof** [comment]
There should be no need to copy any OpenNi2 libs to /usr/lib.  Also, it seems like the libfreenect installation may be causing problems.  

Which Kinect sensor are you using?

**Comment by rossanag** [comment]
I am using Kinect 1414, it works fine with OpenNI 1.

Very likey you are right, there's no need to copy the files, but some people with the same problem solved it doing it in that way.

**Comment by rossanag** [comment]
I want to point out the the libfreenect salmple glview always worked, the problem was with the OpenNI2 ones.

Steps I followed besides the previous I mentioned:

1- I reinstalled libusb-1.0, and I noticed the soft link under the it was pointing to a previous version. I updated the soft link
`sudo ln -s /lib/x86_64-linux-gnu/libudev.so.1.13.0 /lib/x86_64-linux-gnu/libudev.so.0`, I also copied the static library under this directory as well since the older one was kept instead.

2- Updated custom.conf file by adding the directories where the OpenNI2, and libfreenect libraries where at
`sudo gedit /etc/ld.so.conf.d/custom.conf`

To get NiViewer working I edited the OpenNI2.ini file and commented out the respository line to make it pointing to where the libraries are:
`Repository=/home/studio/OpenNI2/Tools`
It works with relatives paths also.

I  don't know if it's the best way to get the things set, at least the examples are working now, any suggestions to an improvement is welcome though.

Best!
